$MirOS$

	Add a way to avoid emitting __stack_chk_fail_local refs

--- gcc/targhooks.c.orig	Fri Feb 20 15:20:14 2009
+++ gcc/targhooks.c	Thu May 29 22:56:06 2014
@@ -500,7 +500,7 @@ default_external_stack_protect_fail (voi
 tree
 default_hidden_stack_protect_fail (void)
 {
-#ifndef HAVE_GAS_HIDDEN
+#if !defined(HAVE_GAS_HIDDEN) || defined(TARGET_LIBC_PROVIDES_SSP_NONLOCAL_ONLY)
   return default_external_stack_protect_fail ();
 #else
   tree t = stack_chk_fail_decl;
